schematic/manifest/generator.py

```diff
@@ -1489,6 +1490,107 @@ def _handle_output_format_logic(self, output_format: str = None, output_path: st
         # Default return a DataFrame
         else:
             return dataframe
+    
+    @staticmethod
+    def create_single_manifest(jsonld: str, data_type: str, access_token:str=None, dataset_id:str=None, strict:bool=True, title:str=None, output_format:str=None, use_annotations:bool=False) -> Union[str, pd.DataFrame, BinaryIO]:
+        """Create a single manifest
+
+        Args:
+            jsonld (str): jsonld schema 
+            data_type (str): data type of a manifest
+            access_token (str, optional): synapse access token. Required when getting an existing manifest. Defaults to None.
+            dataset_id (str, optional): dataset id when generating an existing manifest. Defaults to None.
+            strict (bool, optional): strictness with which to apply validation rules to google sheets. Defaults to True.
+            title (str, optional): title of a given manifest. Defaults to None.
+            output_format (str, optional): format of manifest. It has three options: google sheet, excel or dataframe. Defaults to None.
+            use_annotations (bool, optional): whether to use annotations. Defaults to False.
+
+        Returns:
+            Union[str, pd.DataFrame, BinaryIO]: Googlesheet URL or pandas dataframe or Excel.
+        """
+        # create object of type ManifestGenerator
+        manifest_generator = ManifestGenerator(
+            path_to_json_ld=jsonld,
+            title=title,
+            root=data_type,
+            use_annotations=use_annotations,
+            alphabetize_valid_values = 'ascending',
+        )
+
+        # if returning a dataframe
+        if output_format:
+            if "dataframe" in output_format:
+                output_format = "dataframe"
+
+        result = manifest_generator.get_manifest(
+            dataset_id=dataset_id, sheet_url=True, output_format=output_format, access_token=access_token, strict=strict,
+        )
+
+        # return an excel file if output_format is set to "excel"
+        if output_format == "excel":
+            dir_name = os.path.dirname(result)
+            file_name = os.path.basename(result)
+            mimetype='application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
+            return send_from_directory(directory=dir_name, path=file_name, as_attachment=True, mimetype=mimetype, max_age=0)
+               
+        return result
+    
+    @staticmethod
+    def create_manifests(jsonld:str, data_types:list, access_token:str=None, dataset_ids:list=None, output_format:str=None, title:str=None, strict:bool=None, use_annotations:bool=False) -> Union[str, pd.DataFrame, BinaryIO]:
+        """Create multiple manifests
```

Comment:
All the comments above to `create_single_manifest` apply here as well.

Reply:
Shouldn't `-> Union[str, pd.DataFrame, BinaryIO]:` be a list?

Reply:
Thank you @andrewelamb . Yes. This should be a list, except `BinaryIO` because currently the APIs could only return one file.

Reply:
Thanks @andrewelamb . Here I updated the typing of `output_format` to align with `create_single_manifest` function, and i also updated `strict:bool=True`